app-modules/service-management/src/Filament/Resources/ProductResource/Pages/ManageProductLicenses.php

```diff
@@ -0,0 +1,189 @@
+    {
+        return $form
+            ->schema([
+                TextInput::make('license')
+                    ->label('License')
+                    ->string()
+                    ->required(),
+                Select::make('assigned_to')
+                    ->relationship(
+                        name: 'contact',
+                        titleAttribute: 'full_name',
+                    )
+                    ->label('Assigned To')
+                    ->native(false)
+                    ->preload()
+                    ->searchable(),
+                Textarea::make('description')
+                    ->label('Description')
+                    ->string()
+                    ->columnSpanFull(),
+                DatePicker::make('start_date')
+                    ->label('Start Date')
+                    ->displayFormat('d-m-Y')
+                    ->closeOnDateSelection()
+                    ->required()
+                    ->native(false)
+                    ->live()
+                    ->afterStateUpdated(
+                        fn (Get $get, Set $set) => $get('start_date') > $get('expiration_date') ? $set('expiration_date', '') : ''
+                    ),
+                Group::make([
+                    Checkbox::make('license_does_not_expire')
+                        ->label('License Does Not Expire')
+                        ->live()
+                        ->afterStateHydrated(function (Set $set, $state, $record) {
+                            if ($record && is_null($record->expiration_date)) {
+                                $set('license_does_not_expire', true);
+                            }
+                        }),
+                    DatePicker::make('expiration_date')
+                        ->label('Expiration Date')
+                        ->displayFormat('d-m-Y')
+                        ->closeOnDateSelection()
+                        ->required(fn (Get $get) => ! $get('license_does_not_expire'))
+                        ->disabled(fn (Get $get) => $get('license_does_not_expire'))
+                        ->minDate(fn (Get $get) => $get('start_date'))
+                        ->after('start_date')
+                        ->live(onBlur: true)
+                        ->native(false),
+                ]),
+            ]);
+    }
+
+    public function table(Table $table): Table
+    {
+        return $table
+            ->columns([
+                MaskColumn::make('license')
+                    ->label('License')
+                    ->sortable(),
+                TextColumn::make('contact.full_name')
+                    ->label('Assigned To'),
+                TextColumn::make('start_date')
+                    ->label('Start Date')
+                    ->sortable()
+                    ->dateTime('d-m-Y'),
+                TextColumn::make('expiration_date')
+                    ->label('Expiration Date')
+                    ->sortable()
+                    ->dateTime('d-m-Y'),
```

Comment:
This should probably have a default of a string "No Expiration"

Reply:
Done.